### PR TITLE
sr-iov related to RHOSP (SDN-1170)

### PR DESCRIPTION
**Pull request by mikemckiernan**
* Original PR from Zenghui.
* Updates to terminology.
* Feedback from Aaron about nicSelector hierarchy.

This is a followup PR to #28418.

### CODE REVIEW

**Comment by mikemckiernan** [comment]
@jboxman, @atyronesmith, @zshi-redhat , this is a followup PR to Zenghui's work.  PTAL.

Jason, can you add branch/enterprise-4.7 and Future-Release labels?

File: modules/nw-sriov-networknodepolicy-object.adoc [hunk 1]
```diff
@@ -27,62 +27,56 @@ spec:
...
+When `linkType` is set to `ib`, `isRdma` is automatically set to `true` by an SR-IOV Network Operator webhook. When `linkType` is set to `ib`, `deviceType` should not be set to `vfio-pci`.
 
 [id="sr-iov-network-node-configuration-examples_{context}"]
 == SR-IOV network node configuration examples
 
-The following example describes the configuration for an IB device:
+The following example describes the configuration for an InfiniBand device:
 
-.Example configuration for an IB device
+.Example configuration for an IniniBand device
```
Comment (zshi-redhat) [inline modules/nw-sriov-networknodepolicy-object.adoc:79]: nit: IniniBand -> InfiniBand.

Comment (mikemckiernan) [reply modules/nw-sriov-networknodepolicy-object.adoc:79]: Ick.  Getting the spellchecker for tech terms is on the todo list.  Thank you.

**Comment by mikemckiernan** [comment]
@openshift/team-documention, PTAL.  I favored general reference format for the API objects instead of the object reference format--please let me know if I poisoned it.

**Review by jboxman — COMMENTED** [review]
Left a few comments for review.

File: modules/nw-sriov-networknodepolicy-object.adoc [hunk 2]
```diff
@@ -27,62 +27,58 @@ spec:
...
+<5> Optional: The priority is an integer value between `0` and `99`. A smaller value receives higher priority. For example, a priority of `10` is a higher priority than `99`. The default value is `99`.
 
-<6> Optional: The maximum transmission unit (MTU) of the virtual function. The maximum MTU value can vary for different NIC models.
+<6> Optional: The maximum transmission unit (MTU) of the virtual function. The maximum MTU value can vary for different network interface controller (NIC) models.
 
-<7> The number of the virtual functions (VF) to create for the SR-IOV physical network device. For an Intel Network Interface Card (NIC), the number of VFs cannot be larger than the total VFs supported by the device. For a Mellanox NIC, the number of VFs cannot be larger than `128`.
+<7> The number of the virtual functions (VF) to create for the SR-IOV physical network device. For an Intel network interface controller (NIC), the number of VFs cannot be larger than the total VFs supported by the device. For a Mellanox NIC, the number of VFs cannot be larger than `128`.
```
[the remark this answers is not in the document]
Comment (mikemckiernan) [reply modules/nw-sriov-networknodepolicy-object.adoc:48]: I honestly do not recall where I found the definition.  We one source within Red Hat: https://access.redhat.com/documentation/en-us/red_hat_enterprise_virtualization/3.6/html/technical_reference/network_interface_controller_nic

There's also the output from `lspci` after logging onto a node with `oc debug node/...`.  That returns the following:

```
00:03.0 Ethernet controller: Red Hat, Inc. Virtio network device
```

[the remark this answers is not in the document]
Comment (jboxman) [reply modules/nw-sriov-networknodepolicy-object.adoc:48]: Also: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/configuring_and_managing_networking/index

Also: https://en.wikipedia.org/wiki/Network_interface_controller

I asked in Word Nerds as well.

**Pull request merged**